From 0c5a9d7effa6a22cbe296466dbe3bdced46966ee Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Mon, 14 May 2018 01:09:19 +0200
Subject: [PATCH 2/3] PhoneInterfaceManager: Simplify permission checks for
 endCallForSubscriber()

* Instead of checking the permission ourselves, use the
  nice helper method enforceModifyPermission() to do it

Change-Id: Iabf3b4a75b4362c1ccac419f784a8b22b0742219
---
 src/com/android/phone/PhoneInterfaceManager.java | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/src/com/android/phone/PhoneInterfaceManager.java b/src/com/android/phone/PhoneInterfaceManager.java
index 36d84b70b..c16b9f84b 100644
--- a/src/com/android/phone/PhoneInterfaceManager.java
+++ b/src/com/android/phone/PhoneInterfaceManager.java
@@ -71,7 +71,6 @@ import android.telephony.UssdResponse;
 import android.telephony.VisualVoicemailSmsFilterSettings;
 import android.text.TextUtils;
 import android.util.ArraySet;
-import android.util.EventLog;
 import android.util.Log;
 import android.util.Pair;
 import android.util.Slog;
@@ -1210,12 +1209,7 @@ public class PhoneInterfaceManager extends ITelephony.Stub {
      * @return true is a call was ended
      */
     public boolean endCallForSubscriber(int subId) {
-        if (mApp.checkCallingOrSelfPermission(permission.MODIFY_PHONE_STATE)
-                != PackageManager.PERMISSION_GRANTED) {
-            Log.i(LOG_TAG, "endCall: called without modify phone state.");
-            EventLog.writeEvent(0x534e4554, "67862398", -1, "");
-            throw new SecurityException("MODIFY_PHONE_STATE permission required.");
-        }
+        enforceModifyPermission();
         return (Boolean) sendRequest(CMD_END_CALL, null, new Integer(subId));
     }
 
-- 
2.17.0

